tools/src/bin/rust-indexer.rs

```diff
@@ -281,6 +416,24 @@ fn read_existing_contents(
     }
 }
 
+/// Given a reader and a span from that file, extract the text contained by the span.  If the span
+/// covers multiple lines, then whatever newline delimiters the file has will be included.
+fn extract_span_from_source(
+    reader: &mut File,
+    span: &data::SpanData,
+) -> String {
+    reader.seek(std::io::SeekFrom::Start(span.byte_start.into())).unwrap();
+    let len = (span.byte_end - span.byte_start) as usize;
+    let mut buffer: Box<[u8]> = vec![0; len].into_boxed_slice();
+    match reader.read_exact(&mut buffer) {
+        Ok(_) => String::from_utf8(buffer.to_vec()).unwrap(),
+        Err(e) => {
+            println!("ERROR: Unable to read file: {:?}", e);
```

Comment:
This should probably `error!(`.

Reply:
This actually is somewhat interesting because a problem I had last time I was trying to do some searchfox hacking was that vagrant was constantly giving me short reads.  But agreed on consistency and that silent errors on short reads aren't helpful either.